polar-core/src/polar.rs

```diff
@@ -249,6 +249,16 @@ impl Polar {
             kb.clear_rules();
             return Err(e);
         }
+
+        // TODO(@gkaemmer) is this the right place for this?
```

Comment:
I can't think of a better place really, but at some point we should probably pull our errors/warnings into a shared pipeline and data structure

Reply:
Yeah, that's kinda why I was thinking this was the best place for now. If there are other errors during resource block rewriting/validation, then this check will not be performed. That being said, I think we do always run `check_messages` regardless of whether or not the FFI call returns an error response.

On your last question, I added some thoughts to another comment that might address it.